clients/src/test/java/org/apache/kafka/clients/producer/internals/SenderTest.java

```diff
@@ -467,6 +470,560 @@ public void testClusterAuthorizationExceptionInInitProducerIdRequest() throws Ex
         assertSendFailure(ClusterAuthorizationException.class);
     }
 
+
+    @Test
+    public void testIdempotenceWithMultipleInflights() throws Exception {
+        final long producerId = 343434L;
+        TransactionManager transactionManager = new TransactionManager();
+        setupWithTransactionState(transactionManager);
+        prepareAndReceiveInitProducerId(producerId, Errors.NONE);
+        assertTrue(transactionManager.hasProducerId());
+
+        assertEquals(0, transactionManager.sequenceNumber(tp0).longValue());
+
+        // Send first ProduceRequest
+        Future<RecordMetadata> request1 = accumulator.append(tp0, time.milliseconds(), "key".getBytes(), "value".getBytes(), null, null, MAX_BLOCK_TIMEOUT).future;
```

Comment:
I think we should be asserting on the result of the send futures in these test cases.

Reply:
Just added the assertions for test cases in which they were missing.